projects/plugins/boost/docs/DEVELOPEMENT_GUIDE.md

```diff
@@ -0,0 +1,352 @@
+# Development guide
```

Comment:
I wonder if we need all this, given that most of it refers to the monorepo already?

Reply:
Again here I was debating too.  My aim with it was to be a little bit more specific to Boost development even if some of the content is extremely related to the monorepo documentation. I am happy to take most of it out if it makes more sense. 

I'll let @thingalon chip in too when he got the opportunity to review this PR.

Reply:
I think it would be better to just link back to the main Jetpack Monorepo documentation wherever possible instead of replicating it. It means that changes to the Monorepo's processes don't have to be copied / propegated into Boost later. :)

Reply:
👍

> In fact, those instructions (e.g.: "here is how you just watch this specific plugin") may be worth adopting into the main Jetpack docs as they may help new people get started with other Jetpack-repos plugins?

Yeah, that'd be nice. `docs/development-environment.md` is starting to get very long, and maybe we could break it down and have a more expanded doc on just the `Development workflow` part of it, where we'd expand more on such things.

Reply:
With commit https://github.com/Automattic/jetpack/pull/20669/commits/3acce156eba89b6c8f376e69ccb711705bb41c88, I have now greatly simplified the Jetpack Boost documentation and kept the strict minimum that is a little more relevant to Jetpack Boost. I hope it still makes sense as it is now.

I did not take on the task in this PR to `overhaul` the `docs/development-environment.md` documentation as it is bigger than this PR and I might not be the best person to do it as I am not 100% acquainted to the whole Jetpack Monorepo. (thought I am happy to take a shot at it in another PR if we want to tackle this :) )